### PR TITLE
Add spec for Blacklight::SearchState#filters

### PR DESCRIPTION
**Pull request by dkinzer**
Documents via spec the `filters` method with regards to configured pivot
fields.

REF #2463

### CODE REVIEW

File: spec/lib/blacklight/search_state_spec.rb [hunk 1]
```diff
@@ -402,4 +402,46 @@
       expect(search_state.url_for_document(doc)).to eq doc
     end
   end
+
+  describe '#filters' do
+    context 'pivot facet config but no facet params' do
+      let(:params) { {} }
+
+      it 'maps no filters' do
+        blacklight_config.add_facet_field 'some_key', pivot: %w[pivot_field_1 pivot_field_2]
```
Comment (jcoyne) [inline spec/lib/blacklight/search_state_spec.rb:411]: Can you move this to a `before` block?

File: spec/lib/blacklight/search_state_spec.rb [hunk 2]
```diff
@@ -402,4 +402,46 @@
...
+      end
+    end
+
+    context 'Fully configured pivot facets and some pivot facet params' do
+      let(:params) { { 'f' => { 'pivot_field_1' => 'foo' } } }
+
+      it 'maps the pivot matching param pivot_field' do
+        blacklight_config.add_facet_field "pivot_field_1", show: false
+        blacklight_config.add_facet_field "pivot_field_2", show: false
+        blacklight_config.add_facet_field 'some_key', pivot: %w[pivot_field_1 pivot_field_2]
```
Comment (jcoyne) [inline spec/lib/blacklight/search_state_spec.rb:422]: Can you move these to a before block?

File: spec/lib/blacklight/search_state_spec.rb [hunk 3]
```diff
@@ -402,4 +402,46 @@
...
+        expect(search_state.filters.first.key).to eq('pivot_field_1')
+        expect(search_state.filters.first.values).to eq(['foo'])
+      end
+    end
+
+    context 'Not fully configured pivot facets and some pivot facet params' do
+      let(:params) { { 'f' => { 'pivot_field_1' => 'foo' } } }
+
+      it 'maps no filters' do
+        blacklight_config.add_facet_field 'some_key', pivot: %w[pivot_field_1 pivot_field_2]
```
Comment (jcoyne) [inline spec/lib/blacklight/search_state_spec.rb:432]: Can you move this to a `before` block?

File: spec/lib/blacklight/search_state_spec.rb [hunk 4]
```diff
@@ -402,4 +402,46 @@
...
+        expect(search_state.filters).to eq([])
+      end
+    end
+
+    context 'regular facet config with params' do
+      let(:params) { { 'f' => { 'some_key' => 'foo' } } }
+
+      it 'maps the field matching param facet_field' do
+        blacklight_config.add_facet_field 'some_key'
```
Comment (jcoyne) [inline spec/lib/blacklight/search_state_spec.rb:441]: Can you move this to a `before` block?

File: spec/lib/blacklight/search_state_spec.rb [hunk 5]
```diff
@@ -402,4 +402,46 @@
       expect(search_state.url_for_document(doc)).to eq doc
     end
   end
+
+  describe '#filters' do
+    context 'pivot facet config but no facet params' do
```
Comment (jcoyne) [inline spec/lib/blacklight/search_state_spec.rb:407]: What do you think about starting the context with a  "with" or "when" as illustrated by https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ContextWording  I find this style to be very readable.

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/lib/blacklight/search_state_spec.rb (hunk 5 above)
Comment (dkinzer) [reply spec/lib/blacklight/search_state_spec.rb]: Thanks for the review @jcoyne.  I've updated the specs with requested changes.

File: spec/lib/blacklight/search_state_spec.rb (hunk 2 above)
Comment (dkinzer) [reply spec/lib/blacklight/search_state_spec.rb:425]: done

File: spec/lib/blacklight/search_state_spec.rb (hunk 3 above)
Comment (dkinzer) [reply spec/lib/blacklight/search_state_spec.rb:438]: done

File: spec/lib/blacklight/search_state_spec.rb (hunk 4 above)
Comment (dkinzer) [reply spec/lib/blacklight/search_state_spec.rb:450]: done

**Comment by dkinzer** [comment]
I'm closing this because it looks like there will be movement in fixing the root cause of the issue via  #2692

**Pull request closed without merging**